crates/common/tedge_config_macros/examples/multi.rs

```diff
@@ -0,0 +1,66 @@
+use camino::Utf8PathBuf;
+// use std::net::IpAddr;
+// use std::net::Ipv4Addr;
+use std::num::NonZeroU16;
+// use std::path::PathBuf;
+use tedge_config_macros::*;
+
+#[derive(thiserror::Error, Debug)]
+pub enum ReadError {
+    #[error(transparent)]
+    ConfigNotSet(#[from] ConfigNotSet),
+    #[error("Something went wrong: {0}")]
+    GenericError(String),
+    #[error(transparent)]
+    Multi(#[from] tedge_config_macros::MultiError)
+}
+
+pub trait AppendRemoveItem {
+    type Item;
+
+    fn append(current_value: Option<Self::Item>, new_value: Self::Item) -> Option<Self::Item>;
+
+    fn remove(current_value: Option<Self::Item>, remove_value: Self::Item) -> Option<Self::Item>;
+}
+
+impl<T> AppendRemoveItem for T {
+    type Item = T;
+
+    fn append(_current_value: Option<Self::Item>, _new_value: Self::Item) -> Option<Self::Item> {
+        unimplemented!()
+    }
+
+    fn remove(_current_value: Option<Self::Item>, _remove_value: Self::Item) -> Option<Self::Item> {
+        unimplemented!()
+    }
+}
+
+#[allow(dead_code)]
+define_tedge_config! {
+    #[tedge_config(multi)]
+    c8y: {
+        #[tedge_config(reader(private))]
+        url: String,
+    },
+}
+
+fn main() {
+    // let dto = TEdgeConfigDto::default();
```

Comment:
Same as above, I should replace this with something that actually demonstrates the functionality.